raiden/tests/integration/transfer/test_mediatedtransfer.py

```diff
@@ -210,21 +221,41 @@ def test_mediated_transfer_messages_out_of_order(
     )
 
     amount = 10
-    mediated_transfer(
-        app0,
-        app2,
+    identifier = 1
+    transfer_received = app0.raiden.start_mediated_transfer_with_secret(
         token_network_identifier,
         amount,
-        timeout=network_wait * number_of_nodes,
+        app2.raiden.address,
+        identifier,
+        secret,
     )
 
+    # - Wait until reveal secret is received to replay the message
+    # - The secret is revealed backwards, app2 should be first
+    # - The locked transfer is sent before the secret reveal, so the mediated
+    #   transfers async results must be set and `get_nowait` can be used
+    app2_revealsecret.get(timeout=network_wait)
+    mediated_transfer_msg = app2_mediatedtransfer.get_nowait()
+    app2.raiden.message_handler.handle_message_lockedtransfer(
+        app2.raiden,
+        mediated_transfer_msg,
+    )
+
+    app1_revealsecret.get(timeout=network_wait)
+    app1.raiden.message_handler.handle_message_lockedtransfer(
+        app1.raiden,
+        app1_mediatedtransfer.get_nowait(),
+    )
```

Comment:
Should we hook into the event handler and assert that the repeated lockedtransfer was rejected here (Should be `EventUnlockClaimFailed` looking at the `handle_inittarget` code)? what i am suggesting is probably done somewhere else but i guess it wouldn't hurt checking here as well.

Reply:
That would change the test, I'm not 100% what was the property tested since it lacked documentation.

Because the current test dispatches the locked transfer after the secret reveal, and not after the unlock, then we can't assert for the presence of the unlock claim failed.